src/filerepository.js

```diff
@@ -0,0 +1,455 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module upload/filerepository
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+
+import CKEditorError from '@ckeditor/ckeditor5-utils/src/ckeditorerror';
+import ObservableMixin from '@ckeditor/ckeditor5-utils/src/observablemixin';
+import Collection from '@ckeditor/ckeditor5-utils/src/collection';
+import mix from '@ckeditor/ckeditor5-utils/src/mix';
+import log from '@ckeditor/ckeditor5-utils/src/log';
+
+import FileReader from './filereader.js';
+
+import uid from '@ckeditor/ckeditor5-utils/src/uid.js';
+
+/**
+ * FileRepository plugin.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class FileRepository extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get pluginName() {
+		return 'fileRepository';
```

Comment:
That's a good question :P We'll have to propose a convention there too, so contrib stuff doesn't conflict. But you're right that it should be the same convention.

We have two choices – to either say that only our stuff doesn't have to be prefixed. Or to always prefix our stuff with `ckeditor` or `cksource`. This will look ridiculous though if you have `ckeditor/bold` or something. I'll open a ticket to review this generally. For now, just please follow the current convention.